app/models/manageiq/providers/inventory.rb

```diff
@@ -6,26 +6,13 @@ class Inventory
 
     attr_accessor :collector, :parsers, :persister
 
-    # Based on the given provider/manager class, this returns correct parser class
-    #
-    # @param klass class of the Provider/Manager
-    # @return [Class] Correct class name of the Parser
-    def self.parser_class_for(klass)
-      provider_module = ManageIQ::Providers::Inflector.provider_module(klass)
-      "#{provider_module}::Inventory::Parser::#{klass.name.demodulize}".safe_constantize
-    rescue ManageIQ::Providers::Inflector::ObjectNotNamespacedError => _err
-      nil
-    end
-
-    # Based on the given provider/manager class, this returns correct persister class
-    #
-    # @param klass class of the Provider/Manager
-    # @return [Class] Correct class name of the persister
-    def self.persister_class_for(klass)
-      provider_module = ManageIQ::Providers::Inflector.provider_module(klass)
-      "#{provider_module}::Inventory::Persister::#{klass.name.demodulize}".safe_constantize
-    rescue ManageIQ::Providers::Inflector::ObjectNotNamespacedError => _err
-      nil
+    # Entry point for building inventory
+    def self.build(ems, target)
+      new(
+        class_for(ems, target, 'Persister').new(ems, target),
+        class_for(ems, target, 'Collector').new(ems, target),
```

Comment:
Instead of the general `class_for` here how about we call `collector_class_for` and `persister_class_for` here to give providers a change to override them?